src/Combinatorics/Matroids/properties.jl

```diff
@@ -0,0 +1,1040 @@
+    is_regular, is_binary, is_ternary,
+    n_connected_components, connected_components, is_connected,
+    loops, coloops, is_loopless, is_coloopless, is_simple, direct_sum_components,
+    connectivity_function, is_vertical_k_separation, is_k_separation,
+    vertical_connectivity, girth, tutte_connectivity,
+    tutte_polynomial, characteristic_polynomial, charpoly, reduced_characteristic_polynomial,
+    revlex_basis_encoding, isisomorphic
+
+################################################################################
+##  Properties and basic functions
+################################################################################
+
+
+@doc Markdown.doc"""
+    isomorphism(M::matroid, gs::AbstractVector)
+
+Return a matroid isomorphic to `M` on the groundset `gs`.
+
+# Example
+```jldoctest
+julia> isomorphism(fano_matroid(), [2,3,4,1,5,6,7])
+Matroid of rank 3 on 7 elements
+```
+"""
+function isomorphism(M::Matroid, gs::GroundsetType)
+    if length(M.groundset)!=length(gs)
+        error("Sets of different size")
+    end
+    gs2num = create_gs2num(gs)
+    return Matroid(M.pm_matroid,gs,gs2num)
+end
+
+
+@doc Markdown.doc"""
+    size_groundset(M::matroid)
+
+Return the size of the ground set of the matroid `M`.
+
+# Example
+```jldoctest
+julia> size_groundset(fano_matroid())
+7
+```
+"""
+size_groundset(M::Matroid) = length(M.groundset)
+
+@doc Markdown.doc"""
+    bases(M::matroid)
+
+Return the list of bases of the matroid `M`.
+
+# Example
+```jldoctest
+julia> bases(uniform_matroid(2, 3))
+3-element Vector{Vector{Int64}}:
+ [1, 2]
+ [1, 3]
+ [2, 3]
+```
+"""
+bases(M::Matroid) = [[M.groundset[i+1] for i in sort(collect(C))] for C in Vector{Set{Int}}(M.pm_matroid.BASES)]
+
+@doc Markdown.doc"""
+    nonbases(M::matroid)
+
+Return the list of nonbases of the matroid `M`.
+
+# Example
+```jldoctest
+julia> nonbases(fano_matroid())
+7-element Vector{Vector{Int64}}:
+ [1, 2, 3]
+ [1, 4, 5]
+ [1, 6, 7]
+ [2, 4, 6]
+ [2, 5, 7]
+ [3, 4, 7]
+ [3, 5, 6]
+ 
+```
+"""
+nonbases(M::Matroid) = [[M.groundset[i+1] for i in N] for N in M.pm_matroid.NON_BASES] 
+
+
+@doc Markdown.doc"""
+    circuits(M::matroid)
+
+Return the list of circuits of the matroid `M`.
+
+# Example
+```jldoctest
+julia> circuits(uniform_matroid(2, 4))
+4-element Vector{Vector{Int64}}:
+ [1, 2, 3]
+ [1, 2, 4]
+ [1, 3, 4]
+ [2, 3, 4]
+```
+"""
+circuits(M::Matroid) = [[M.groundset[i+1] for i in sort(collect(C))] for C in Vector{Set{Int}}(M.pm_matroid.CIRCUITS)]
+
+@doc Markdown.doc"""
+    hyperplanes(M::matroid)
+
+Return the list of hyperplanes of the matroid `M`.
+
+# Example
+```jldoctest
+julia> hyperplanes(fano_matroid())
+7-element Vector{Vector{Int64}}:
+ [3, 5, 6]
+ [3, 4, 7]
+ [2, 5, 7]
+ [2, 4, 6]
+ [1, 6, 7]
+ [1, 4, 5]
+ [1, 2, 3]
+```
+"""
+hyperplanes(M::Matroid) = [[M.groundset[i+1] for i in sort(collect(C))] for C in Vector{Set{Int}}(M.pm_matroid.MATROID_HYPERPLANES)]
+
+@doc Markdown.doc"""
+    flats(M::matroid, r::Union{Int,Nothing})
```

Comment:
This may be just me, but since the use of `nothing` here is really an implementation detail that shouldn't concern the user, perhaps it'd be nicer to document this like this:
```suggestion
    flats(M::matroid)
    flats(M::matroid, r::Int)
```

Reply:
The same for `cyclic_flats` of course